File: src/Application Specific/Classes/Entity Collections/Documents/DocumentsValidator.vb
Namespace Entity

    Namespace Documentss

        Public Class DocumentsValidator
            'make sure that contact object is valid
            Inherits BaseValidator

            Public Sub Validate(ByVal oDocuments As Documents)

                'make sure that contact object is valid
                If oDocuments.Errors.Count > 0 Then
                    Dim de As DictionaryEntry
                    For Each de In oDocuments.Errors
                        Me.AddCriticalMessage(CStr(de.Value))
                    Next
                End If

                If oDocuments.TableEnumID = CInt(Entity.Sys.Enums.TableNames.NO_TABLE) Then
                    Me.AddCriticalMessage("Document Entity Missing")
                End If
                If oDocuments.RecordID = 0 Then
                    Me.AddCriticalMessage("Document Record ID Missing")
                End If
                If oDocuments.DocumentTypeID = 0 Then
                    Me.AddCriticalMessage("Document Type Missing")
                End If
                If oDocuments.Name.Trim.Length = 0 Then
                    Me.AddCriticalMessage("Document Reference Missing")
                End If
                If oDocuments.Location.Trim.Length = 0 Then
                    Me.AddCriticalMessage("Document Missing")
                End If

                If Me.ValidatorMessages.CriticalMessages.Count > 0 Then
                    Throw New ValidationException(Me)
                End If

            End Sub

        End Class

    End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/Documents/DocumentsValidator.cs
using System.Collections;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace Documentss
    {
        // make sure that contact object is valid
        public class DocumentsValidator : BaseValidator
        {
            public void Validate(Documents oDocuments)
            {

                // make sure that contact object is valid
                if (oDocuments.Errors.Count > 0)
                {
                    DictionaryEntry de;
                    foreach (var de in oDocuments.Errors)
                        AddCriticalMessage(Conversions.ToString(de.Value));
                }

                if (oDocuments.TableEnumID == Conversions.ToInteger(Sys.Enums.TableNames.NO_TABLE))
                {
                    AddCriticalMessage("Document Entity Missing");
                }

                if (oDocuments.RecordID == 0)
                {
                    AddCriticalMessage("Document Record ID Missing");
                }

                if (oDocuments.DocumentTypeID == 0)
                {
                    AddCriticalMessage("Document Type Missing");
                }

                if (oDocuments.Name.Trim().Length == 0)
                {
                    AddCriticalMessage("Document Reference Missing");
                }

                if (oDocuments.Location.Trim().Length == 0)
                {
                    AddCriticalMessage("Document Missing");
                }

                if (ValidatorMessages.CriticalMessages.Count > 0)
                {
                    throw new ValidationException(this);
                }
            }
        }
    }
}